ui_components/src/chat/Chat_new.js

```diff
@@ -0,0 +1,385 @@
+    const charactersRemainingDisplay = `Characters remaining ${charactersRemaining}`;
+
+    return (
+        <div className="chat_input_container">
+            {showMemeSelect &&
+                <ChatMemeModal
+                    memes={memes}
+                    selectMeme={handleMemeSelect}
+                    closeMemeSelect={() => setShowMemeSelect(false)}
+                />
+            }
+            <div className="chat_input_left">
+                <div className="chat_settings_container">
+                    <label className={"quick_reply_label ft-s ft-c1 ft-medium"}>quick-reply</label>
+                    <input className="quick_reply_box" type="checkbox" checked={quickReply} onChange={e => setQuickReply(e.target.checked)} />
+                </div>
+                <div className="chat_meme_toggle_container">
+                    <button className={"meme_toggle button-bar_large ft-s ft-c1 ft-medium t-hover"} onClick={() => setShowMemeSelect(!showMemeSelect)}>meme list</button>
+                </div>
+            </div>
+            <div className="chat_input_center">
+                <textarea
+                    id="chat_input_box"
+                    minLength="3"
+                    maxLength={maxPostLength}
+                    value={message}
+                    onChange={e => setMessage(e.target.value)}
+                    onKeyDown={handleKeyDown}
+                ></textarea>
+                <span className="ft-s ft-c1">{charactersRemainingDisplay}</span>
+            </div>
+            <div className="chat_input_right">
+                <button className={"chat_submit_button button-bar_large ft-s ft-c1 ft-medium t-hover"} onClick={handlePostSubmit}>post</button>
+            </div>
+        </div>
+    )
+}
+
+function ChatMemeModal({ memes, selectMeme, closeMemeSelect }) {
+    return (
+        <div id="meme_modal">
+            <div className="meme_modal_header">
+            </div>
+            <div>
+                <div id="meme_box">
+                    {memes.codes.map((meme, i) => (
+                        <div key={`meme:${i}`} className="meme_select">
+                            <img src={memes.urls[i]} onClick={() => selectMeme(i)} />
+                        </div>
+                    ))}
+                </div>
+            </div>
+            <div className="meme_modal_footer">
+                <button className={"meme_close button-bar_large ft-s ft-c1 ft-medium t-hover"} onClick={closeMemeSelect}>close</button>
+            </div>
+        </div>
+    );
+}
+
+function ChatPosts({
+    posts,
+    previousPagePostId,
+    nextPagePostId,
+    isModerator,
+    deletePost,
+    quotePost,
+    goToPreviousPage,
+    goToNextPage
+}) {
+    return (
+        <>
+            <div id="chat_navigation">
+                <div className="chat_navigation_divider_left"><svg width="100%" height="2"><line x1="0%" y1="1" x2="100%" y2="1" stroke="#4e4535" strokeWidth="1"></line></svg></div>
```

Comment:
```suggestion
                <div className="chat_navigation_divider_left">
                  <svg width="100%" height="2"><line x1="0%" y1="1" x2="100%" y2="1" stroke="#4e4535" strokeWidth="1"></line></svg>
                </div>
```

Reply:
Style-wise, would be great to break these lines up a little bit between the container and contents